src/components/App/Component.js

```diff
@@ -0,0 +1,31 @@
+import React, { PureComponent, PropTypes } from 'react'
+import { createMemoryHistory, browserHistory } from 'react-router'
+import { fetchConfig } from 'store/config'
+import { fetchUser } from 'store/session'
+
+const history = browserHistory || createMemoryHistory()
+
+class App extends PureComponent {
+  static propTypes = {
+    user: PropTypes.object,
+    children: PropTypes.node
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { user } = this.props
+    if (nextProps.user !== user) {
+      nextProps.user ? history.replace('/') : history.replace('/login')
```

Comment:
yeah, maybe we can figure out a way to easily enable/disable it. It's going to be needed in most apps, so I don't want to make it a pain in the ass to add each time.